pkg/testing/integration/program.go

```diff
@@ -292,6 +293,12 @@ type ProgramTestOptions struct {
 	UseAutomaticVirtualEnv bool
 	// Use the Pipenv tool to manage the virtual environment.
 	UsePipenv bool
+	// Use a shared virtual environment for tests based on the contents of the requirements file. Defaults to false.
+	UseSharedVirtualEnv *bool
```

Comment:
I really don't love that I have to use a pointer here, but it _seems_ to be the only way to disambiguate between `false` and an unset value at initialization time.  I've used a convenience function to try to alleviate the ergonomics issue this introduces, but I'm open to other approaches if anyone has any.

Reply:
I don't have a better answer here either, I expect we'll want to make this the default in the future because it'll speed up examples & templates tests, tests in providers. 

Using a pointer for Maybe[bool] and an env var to set the default is a good approach for a testing harness, and we can change the default in the future after validating.